### DISCUSSION TITLE
How feasible would be Viv as a HTML live preview similar to md?

### DISCUSSION THREAD

**Discussion by tuurep** [jannis-baum/Vivify#84] [Ideas]
Question motivated by often writing a HTML file in nvim and thinking "darn, I wish vivify handled this, too!"

It's unclear to me whether this is totally out of scope, or perhaps an easy addition. What do you think?

**Comment by jannis-baum** [comment]
First thing I was going to say was that this should be a very easy addition. Right now we detect HTML files to be plain text, i.e. we wrap them into a Markdown document and show the code. If skip this and just return the raw HTML, it will display.

But of course it's not quite as simple, unfortunately:

1. We need to inject the `client.js` script into the HTML file so that we still have communication with `vivify-server`, e.g. so that the server stays up knowing it has a client and so that auto-updating from the Vim plugin works. Injecting the script should also be rather straight forward though
2. The part where it will actually get complicated is if you want the HTML to be able to "import" other local files, most notably probably local CSS and JS. For this to work "just like that" we would also have to get rid of the "Markdown code document"-wrapping for these file types, but then we couldn't nicely look at the code anymore, i.e. we would lose another feature. Another thing is that the `client.js` could be broken by another script from the HTML that overwrites global variables.
3. If we just serve the HTML as it is with the injected `client.js` script, we will also not have the "back"-button there. So another option might also be to somehow wrap the HTML, but this would probably be complicated. I don't know how easy it is to display an HTML file as content of another HTML file.

These are my first thoughts on what we would have to figure out to make this work. The simplest solution would be to just remove all scripts/styles from the HTML and serve it with the injected script. Then we would not have point number 2, but I guess then the usefulness would also be voided ^^

**Comment by tuurep** [comment]
Thank you, very interesting
